browser/components/EditProject.tsx

```diff
@@ -110,33 +111,35 @@ export default class EditProject extends React.Component<Props, State> {
   }
 
   async componentDidUpdate(prevProps) {
-    const projectInfo = await this.getProject();
-    const projectGroups = [];
-    for (const group of projectInfo.groups) {
-      projectGroups.push(group.group_id);
-    }
-    const projectUsers = [];
-    let projectDetails = {
-      project_id: null,
-      project_license: null,
-      project_name: null,
-      project_url: null,
-    };
-    if (projectInfo) {
-      projectDetails = projectInfo.project;
-      for (const user of projectInfo.users) {
-        projectUsers.push(user.company_alias);
+    if (this.props !== prevProps) {
+      const projectInfo = await this.getProject();
+      const projectGroups = [];
+      for (const group of projectInfo.groups) {
+        projectGroups.push(group.group_id);
+      }
+      const projectUsers = [];
+      let projectDetails = {
+        project_id: null,
+        project_license: null,
+        project_name: null,
+        project_url: null,
+      };
+      if (projectInfo) {
+        projectDetails = projectInfo.project;
+        for (const user of projectInfo.users) {
+          projectUsers.push(user.company_alias);
+        }
       }
+      this.setState({
```

Comment:
Since you're deriving state from props, you most likely want to just compute these values during render (and not use state at all for these variables). You're risking a render lockup doing this in componentDidUpdate.

Reply:
Like I said above, I was following the pattern from EditGroup but I can refactor if you want.

Reply:
I read https://reactjs.org/docs/react-component.html#componentdidupdate again and as long as I am wrapping setState in a conditional, which I have changed the conditional to trigger on a specific set of data instead of the reference comparison, it should be fine.